crates/context/config/src/client/env/proxy/types/evm.rs

```diff
@@ -117,23 +189,191 @@ impl From<SolProposalAction> for ProposalAction {
+                        name: method_name,
+                        inputs: parsed_args
+                            .iter()
+                            .enumerate()
+                            .map(|(index, (key, _value))| {
+                                let param_type = match key.as_str() {
+                                    "string" => ParamType::String,
+                                    "uint256" => ParamType::Uint(256),
+                                    "int256" => ParamType::Int(256),
+                                    "address" => ParamType::Address,
+                                    "bool" => ParamType::Bool,
+                                    "bytes" => ParamType::Bytes,
+                                    "array(string)" => {
+                                        ParamType::Array(Box::new(ParamType::String))
+                                    }
+                                    "array(uint256)" => {
+                                        ParamType::Array(Box::new(ParamType::Uint(256)))
+                                    }
+                                    "array(int256)" => {
+                                        ParamType::Array(Box::new(ParamType::Int(256)))
+                                    }
+                                    "array(address)" => {
+                                        ParamType::Array(Box::new(ParamType::Address))
+                                    }
+                                    "array(bool)" => ParamType::Array(Box::new(ParamType::Bool)),
+                                    "array(bytes)" => ParamType::Array(Box::new(ParamType::Bytes)),
+                                    "tuple" => ParamType::Tuple(vec![]),
+                                    _ => panic!("Unsupported parameter type: {}", key),
+                                };
+
+                                Param {
+                                    name: format!("param{}", index),
+                                    kind: param_type,
+                                    internal_type: None,
+                                }
+                            })
+                            .collect(),
+                        outputs: vec![],
+                        constant: Some(false),
+                        state_mutability,
+                    };
+
+                    let call_data = function.encode_input(&tokens).unwrap();
+
+                    let contract_address =
+                        Address::from_str(&receiver_id).expect("Invalid address");
+                    let data = SolValue::abi_encode(&(contract_address, call_data, amount));
+
+                    data.into()
+                }
+                ProposalAction::Transfer {
+                    receiver_id,
+                    amount,
+                } => {
+                    let data = TransferData {
+                        recipient: receiver_id.parse().expect("Invalid address"),
```

Comment:
isn't this unsanitized input from the app (untrusted) (which will likely be from some already untrusted frontend)? this would mean a malicious app can crash the node 😕

Reply:
Updated with unwrap_or_else

Reply:
there are others, see `SolProposalAction: From<ProposalAction>`

if there are legitimate cases where the conversion can fail (with no reasonable default), consider making it in to a `TryFrom` instead, so you can return a result